bigquery/v2/query/query.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package query
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/bigquery/v2/apiv2/bigquerypb"
+	"github.com/googleapis/gax-go/v2"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+// Query represents a query job.
```

Comment:
Naming consideration:  Should we avoid using existing terms for custom abstractions? Does "Query" mean more or less to BQ users?

Reply:
+1 if this is a query job, imho it should be named QueryJob - the query is the actual DDL syntax parsed by BQ